feat: replace vm2 sandbox with isolated-vm (#18274)
backstage/backstage: .changeset/popular-donuts-shout.md

---
'@backstage/plugin-scaffolder-backend': patch
---

> Switch out the sandbox to `isolated-vm` in favour of `vm2`.

Reviewer: ```suggestion
Switch out the sandbox, from `vm2` to `isolated-vm`.
```

"in favour of" is the other way around
